src/lib/battery/battery.cpp

```diff
@@ -239,10 +239,10 @@ float Battery::calculateStateOfChargeVoltageBased(const float voltage_v, const f
 void Battery::estimateStateOfCharge()
 {
 	// choose which quantity we're using for final reporting
-	if (_params.capacity > 0.f && _battery_initialized) {
+	if ((_params.capacity > 0.f) && _battery_initialized && (_state_of_charge_volt_based > _params.emergen_thr)) {
 		// if battery capacity is known, fuse voltage measurement with used capacity
 		// The lower the voltage the more adjust the estimate with it to avoid deep discharge
-		const float weight_v = 3e-4f * (1 - _state_of_charge_volt_based);
+		const float weight_v = 3e-2f * (1 - _state_of_charge_volt_based);
```

Comment:
Makes sense to give voltage more weight, especially as it's anyway only fed in when it gets depleted.
Where does this magic number 0.0003 resp 0.03 now come from? Worth a comment in my eyes.

Reply:
These come from testing I did back at Yuneec for the H520 where the previous weight was used on the product because it resulted in the best user experience. With the fine-tuned battery parameters for the single type of battery that was sold with the product the SOC was decreasing pretty linearly until the very end of the flight with light or heavy load, the battery was connected when not full and when the battery already aged a bit.

The 100 times higher weight produced more fluctuations and non-linear decreasing of the SOC but since I have not seen a single user or company spending the extra time at analyzing data and tuning these parameters I suspect it leads to overall better satisfaction.

I should have kept and uploaded the way I was analyzing this, it's been a while but I showed examples here: https://github.com/PX4/PX4-Autopilot/pull/8153

Reply:
How should I put that in a comment? I mean I'm open to any weighting.
Ideally we'd have a script that gives you parameter suggestions based on flight logs and some margin setting.

Reply:
// weighting was determined heuristically
I meant something like that. Just to be clear that this is not linked to any mathematical or physical property.